pandas/_libs/algos.pyx

```diff
@@ -197,9 +195,10 @@ def groupsort_indexer(ndarray[int64_t] index, Py_ssize_t ngroups):
     return result, counts
 
 
+# TODO: redundant with groupby.kth_smallest_c
```

Comment:
its not actually

Reply:
OK.  The typing is more specific in the groupby version, but the code itself is nearly identical

Reply:
I'll get rid of the comment, but am kind of surprised: you're usually Holy Crusader against redundant code, and the _body_ of this function is very nearly copy/pasted